refactor(access): Improve Access Implementation

BeniCheni opened the pull request:
Since #12 is merged for `tink access`, this follow-up PR attempts to improve:
* use `await` to rewrite `.then` syntax
* remove unneeded `.catch` on await code
* address an `TODO` item:
  * call `whoami` when no "user" is provided to `ls-packages` option.

BeniCheni committed:
refactor(access): use await to omit .then syntax

BeniCheni committed:
refactor(access): remove unneeded .catch on await code

BeniCheni committed:
feat(access): call whoami when no user option is provided to ls-packages

zkat commented on lib/commands/access.jsx:
```diff
@@ -96,54 +96,48 @@ const render = (opts, content) => {
   } else {
-    findPrefix(process.cwd()).then(prefix => {
-      readJson(
+    const prefix = await libnpm.findPrefix(process.cwd())
```
It's getPrefix

zkat commented on lib/commands/access.jsx:
```diff
@@ -96,54 +96,48 @@ const render = (opts, content) => {
-      readJson(
+    const prefix = await libnpm.findPrefix(process.cwd())
+    if (prefix) {
+      libnpm.readJson(
```
readJSON*

zkat reviewed (changes requested):
Changes look mostly good -- can you manually test these commands to make sure they're working?

BeniCheni committed:
fix(access): correct libnpm .getPrefix & .readJSON logic

BeniCheni commented:
> ... can you manually test these commands to make sure they're working?

Corrected the bugs you caught in your review comments. Fixed in [commit 3a4f013]()

BeniCheni commented:
> ... can you manually test these commands to make sure they're working?

Corrected the bugs you caught in your review comments. Fixed in [commit 3a4f013](https://github.com/npm/tink/pull/34/commits/3a4f01356f572ff92be68e8b11f95e5c8ffbdbc5). Tested the commands manually to ensure working status, e.g. Thanks for your review! 🙏👍:
```
bchen in ~/github/tink on git:tink-access-refactor [20:41:51]
→ tink access ls-collaborators                           
┌────────┬────────────┐
│ collab │ role       │
├────────┼────────────┤
│ zkat   │ read-write │
└────────┴────────────┘
```

zkat reviewed (approved):
Everything looks good now!
